lib/Service/Gateway/SMS/Provider/PlivoConfig.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @author Chris Jenkins <chris.j@guruinventions.com>
+ *
+ * Plivo - Config for Two-factor Gateway for Plivo
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\TwoFactorGateway\Service\Gateway\SMS\Provider;
+
+
+use function array_intersect;
+use OCA\TwoFactorGateway\AppInfo\Application;
+use OCA\TwoFactorGateway\Exception\ConfigurationException;
+use OCP\IConfig;
+
+class PlivoConfig implements IProviderConfig {
+
+	/** @var IConfig */
+	private $config;
+	
+	public const AUTH_ID_KEY = 'plivo_auth_id';
+	public const AUTH_TOKEN_KEY = 'plivo_auth_token';
+	public const CALLBACK_URL = 'plivo_callback_url';
+	public const SRC_NUMBER_KEY = 'plivo_src_number';
+	
+	private const EXPECTED_KEYS = [
+			self::AUTH_ID_KEY,
+			self::AUTH_TOKEN_KEY,
+			self::CALLBACK_URL,
+			self::SRC_NUMBER_KEY
+		];
+
+	public function __construct(IConfig $config) {
+		$this->config = $config;
+	}
+	
+	private function getInternalValue(string $key): string {
+		$val = $this->config->getAppValue(Application::APP_NAME, $key, null);
+		if (is_null($val)) {
+			throw new ConfigurationException();
+		}
+		return $val;
+	}
+	
+	public function getValue(string $key): string {
+        return $this->getInternalValue($key);
```

Comment:
(*I never thought to write this sentence ANY time in my life... but:*) Please use tab for indenting. It's a coding rule at nc :/

Reply:
Okay, is there a command I can run to fix items as such? For example, `composer cs:fix`  Thanks for letting me know of the coding standard for nc...I'm new.

Reply:
Most IDEs (like PhpStorm, VSCode, etc.) support setting a style and reformat your changes. I not aware of any other way to do this (what doesn't mean there is none ;)).

Reply:
php-cs-fixer I believe does it according to the PSRs.  I used it for my next commit and please let me know if it meets nc's standard.

Reply:
> Most IDEs (like PhpStorm, VSCode, etc.) support setting a style and reformat your changes. I not aware of any other way to do this (what doesn't mean there is none ;)).

I have tried to use Atom, Sublime Text, and VSCode.  All don't seem to like WebDAV.

Reply:
https://github.com/nextcloud/twofactor_gateway/issues/353 for the Nextcloud cs-fixer config